Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -502,20 +502,16 @@ public IEnumerable<FlatHistory> GetHistory(Wallet wallet)
             lock (this.lockObject)
             {
                 // Get transactions contained in the wallet.
-                List<FlatHistory> items = this.GetHistoryInternal(wallet).SelectMany(s => s.Transactions.Select(t => new FlatHistory { Address = s, Transaction = t })).ToList();
+                IEnumerable<FlatHistory> items = this.GetHistoryInternal(wallet).SelectMany(s => s.Transactions.Select(t => new FlatHistory { Address = s, Transaction = t }));
```

Comment:
This will be enumerated outside of the lock right?

Reply:
I would think the same as Dan.
With a lock we protect this piece of code, not the wallet object itself, but what we really want to protect is iterating on the wallet outside of a lock.
It'll be interesting to know how this plays out.